src/NUnitFramework/framework/Internal/Tests/TestAssembly.cs

```diff
@@ -39,25 +39,25 @@ public class TestAssembly : TestSuite
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="TestAssembly"/> class
-        /// specifying the Assembly and the path from which it was loaded.
+        /// specifying the Assembly and the name of the assembly.
         /// </summary>
         /// <param name="assembly">The assembly this test represents.</param>
-        /// <param name="path">The path used to load the assembly.</param>
-        public TestAssembly(Assembly assembly, string path)
-            : base(path)
+        /// <param name="name">The desired name for the test suite.</param>
+        public TestAssembly(Assembly assembly, string name)
+            : base(name)
         {
             this.Assembly = assembly;
-            this.Name = Path.GetFileName(path);
+            this.Name = name;
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="TestAssembly"/> class
-        /// for a path which could not be loaded.
+        /// specifying the name of the assembly.
```

Comment:
This constructor is specifically used when there is no Assembly object because the load failed. Maybe a way to keep reflecting that would be `specifying a suite name for an assembly which failed to load.` or something similar?